aodndata/moorings/products_handler.py

```diff
@@ -0,0 +1,98 @@
+import json
+
+from owslib.fes import PropertyIsEqualTo, PropertyIsNotEqualTo, PropertyIsLike, PropertyIsNotEqualTo, And
+
+from aodncore.pipeline import HandlerBase, PipelineFilePublishType, FileType, PipelineFileCollection, PipelineFile
+from aodncore.pipeline.exceptions import ComplianceCheckFailedError, InvalidFileContentError, InvalidFileNameError
+from aodncore.pipeline.files import RemotePipelineFileCollection
+from aodncore.util.wfs import ogc_filter_to_string
+
+from aodndata.moorings.classifiers import MooringsFileClassifier
+
+
+class MooringsProductClassifier(MooringsFileClassifier):
+    @classmethod
+    def _get_data_category(cls, input_file):
+        return 'aggregated_timeseries'
+
+    @classmethod
+    def _get_product_level(cls, input_file):
+        return ''
+
+
+class MooringsProductsHandler(HandlerBase):
+    """Handler to create products from moorings files.
+
+    The input file is a JSON document containing a site_code and a list of variables. The handler will then create
+    the products for each variable at that site, using all the relevant input files available on S3.
+
+    The structure of the JSON document will be something like this:
+    {
+        "site_code": "NRSMAI",
+        "variables": ["TEMP", "PSAL", "DOX1", "DOX2", "CPHL"]
+    }
+    """
+
+    FILE_INDEX_LAYER = 'imos:moorings_all_map'
+
+    def __init__(self, *args, **kwargs):
+        super(MooringsProductsHandler, self).__init__(*args, **kwargs)
+        if self.allowed_extensions is None:
+            self.allowed_extensions = ['.json']
+        self.product_site_code = None
+        self.product_variables = None
+
+    def preprocess(self):
+        """Collect available input files and create the products, adding them to the collection to be published."""
+
+        # Don't want to publish the manifest file, so remove it from the collection
+        self.file_collection.pop()
```

Comment:
This could be avoided by making it a known file type(e.g. `.product_manifest`) and simply not adding to the collection in the first place

Reply:
:+1: Sure. I'd make it even more generic and call it `.json_manifest`, as it could be used for other purposes.
Or `.manifest.json`, so it has the "correct" extension for a JSON file?

Reply:
The only consideration here really is whether different JSON files should have different behaviour. For example, if in this case we *don't* want it in the collection, but in another case if a JSON file would have real data in it and actually be the "input file", we *would* want it in the collection.

Reply:
As long as it's a _manifest_ file, we generally don't want it in the collection (the handler can add it later anyway).
But as you say, we would not want to do that for just _any_ JSON file, so we do need to distinguish the JSON manifest type from other JSON files.

Reply:
Draft PR here: https://github.com/aodn/python-aodncore/pull/169

Reply:
Thanks, that seems to work.